packages/headless/src/controllers/commerce/product-listing/pagination/headless-product-listing-pagination.test.ts

```diff
@@ -0,0 +1,101 @@
+import {configuration} from '../../../../app/common-reducers';
+import {
+  selectPage,
+  nextPage,
+  previousPage,
+} from '../../../../features/commerce/pagination/pagination-actions';
+import {paginationReducer as pagination} from '../../../../features/commerce/pagination/pagination-slice';
+import {fetchProductListing} from '../../../../features/commerce/product-listing/product-listing-actions';
+import {productListingV2Reducer as productListing} from '../../../../features/commerce/product-listing/product-listing-slice';
+import {buildMockCommerceEngine, MockCommerceEngine} from '../../../../test';
+import {buildMockCommerceState} from '../../../../test/mock-commerce-state';
+import {
+  buildProductListingPagination,
+  ProductListingPagination,
+} from './headless-product-listing-pagination';
+
+describe('ProductListingPagination', () => {
+  let engine: MockCommerceEngine;
+  let productListingPagination: ProductListingPagination;
+
+  function initProductListingPagination(page = 0, totalPages = 1) {
+    const mockState = buildMockCommerceState();
+    engine = buildMockCommerceEngine();
```

Comment:
We can pass a pre-populated state to the engine like this:
```suggestion
    engine = buildMockCommerceEngine({ state: {...} });
```
(instead of setting it as engine property)